jwt integration and fields validation

AfaqAhmed-XGrid opened the pull request:
## **Issue** #13 
     Title: JWT integration with passport and body, params, queries validation

## Description
This pull request brings significant enhancements to our authentication process and data validation. Here's an overview of the changes:
=> Improved Authentication:  I've integrated JSON Web Tokens (JWT) alongside Passport. Once Passport validation is successful, we'll transition to using JWT for secure and efficient authentication and authorization.
=> Data Validation: To ensure the data sent to the backend is robust and secure, I've added comprehensive validation for request bodies, URL parameters, and query strings. This is achieved using express-validator, enabling us to handle input validation more effectively.

## **Checklist**
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the README.md, or the docs directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes (if applicable)
- [x] Every function, interface, class has a comment describing what it does and input/output parameters

AfaqAhmed-XGrid committed:
jwt integration and fields validation

integrated jwt with passport and implement fields validatio fro queries, params and body using express-validator package.

Also corrected some errors in imports categorization

AfaqAhmed-XGrid committed:
made the package version constant

express validator package

AfaqAhmed-XGrid committed:
made the package version constant

express validator package

AfaqAhmed-XGrid committed:
Merge branch 'Improve-passport-integration' of https://github.com/AfaqAhmed-XGrid/Cine-Info-AfaqAhmed-XGrid into jwt-integration-fields-validation

waqar-360 commented on api/modules/auth/auth.controller.js:
```diff
@@ -18,59 +18,103 @@ limitations under the License.
+  passport.authenticate('login', (err, user, info) => {
+    if (err) {
+      logger.error('Passport login Strategy returned error (loginController)', { error: err }, { info: info }, { user: user });
+      return res.status(409).json({ success: 0, message: info?.message, data: { err } });
```
please create separate file for status codes.

AfaqAhmed-XGrid replied:
Done

AfaqAhmed-XGrid committed:
created separate file for response status codes

AfaqAhmed-XGrid committed:
updated validation, cors rules, response status for validation error, swagger

updated password field validation in login endpoint
corrected the reset-password endpoint validation. Used param instead of query
updated recover password pattern
corrected argument of created JwtPayload. Earlier I was giving it wrong user.
updated cors rules for allowing authorization token in header
added response status code in validation error response
updated swagger documentation

waqar-360 commented:
LGTM
